lib/events/dynamoevents/dynamoevents.go

```diff
@@ -446,8 +446,9 @@ func (l *Log) createPutItem(sessionID string, in apievents.AuditEvent) (*dynamod
 		return nil, trace.Wrap(err)
 	}
 	return &dynamodb.PutItemInput{
-		Item:      av,
-		TableName: aws.String(l.Tablename),
+		Item:                av,
+		TableName:           aws.String(l.Tablename),
+		ConditionExpression: aws.String("attribute_not_exists(SessionID) AND attribute_not_exists(EventIndex)"),
```

Comment:
Yes, as soon as they update their auth servers and events from older app servers, they will hit this, at least for the first `app.session.chunk` (which conflicts with `app.session.start`). As mentioned in the PR description, if the app server serving the requests changes, there will be more conflicts (generating more errors).

Reply:
If we release this check with the patch, customers's auth will be "spammed" with this before app servers are upgraded. As discussed offline, it is preferred to handle this separately and ideally release it in a new major version. By that time, app servers should have been upgraded with this patch already and we can likely catch other events that have the same problem during the release testing.

Though, since this check does not block using the app, and we are losing events before app servers are upgraded anyway, I don't mind if we really want to release this together.

Reply:
I prefer having an error log to silently overwriting audit events (current behavior).

@codingllama, Any thoughts on this?

Reply:
I'm afraid I don't have enough context to have a strong opinion here.